manifests/mod/auth_openidc.pp

```diff
@@ -1,10 +1,29 @@
 # @summary
 #   Installs and configures `mod_auth_openidc`.
-# 
+#
+# @param manage_dnf_module Whether to manage the DNF module
+# @param dnf_module_ensure The DNF module name to ensure. Only relevant if manage_dnf_module is set to true.
+# @param dnf_module_name The DNF module name to manage. Only relevant if manage_dnf_module is set to true.
+#
 # @see https://github.com/zmartzone/mod_auth_openidc for additional documentation.
+# @note Unsupported platforms: OracleLinux: 6; RedHat: 6; Scientific: 6; SLES: all
 #
-class apache::mod::auth_openidc inherits apache::params {
+class apache::mod::auth_openidc (
+  Boolean $manage_dnf_module = $facts['os']['family'] == 'RedHat' and $facts['os']['release']['major'] == '8',
```

Comment:
This is correct, because while EL9 has module support, `mod_auth_oidc` is not modular there.
(:exploding_head:)